classes/requests/helpers/class-dintero-checkout-order.php

```diff
@@ -363,16 +368,27 @@ public function get_gift_card( $gift_card ) {
 	 * Formats the shipping method to be used in order.items.
 	 *
 	 * @param WC_Order_Item_Shipping $shipping_item The WooCommerce shipping method.
-	 * @param string                 $package_index An index for uniquely identifying shipping rates that appear multiple times (e.g., in different packages).
 	 * @return array
 	 */
-	public function get_shipping_item( $shipping_item, $package_index = '' ) {
-		$id              = $shipping_item->get_method_id() . ':' . $shipping_item->get_instance_id();
-		$line_id         = $id . ( $package_index !== '' ? ":{$package_index}" : '' );
+	public function get_shipping_item( $shipping_item ) {
+		$id = $shipping_item->get_method_id() . ':' . $shipping_item->get_instance_id();
+
+		// If this is a refund, we need to retrieve the line_id from the non-refund order as the OrderRefund does not contain the original metadata in Order.
+		if ( $this->is_refund ) {
+			$line_id = wc_get_order_item_meta( $shipping_item->get_meta( '_refunded_item_id' ), '_dintero_checkout_line_id', true );
+		} else {
+			$line_id = $shipping_item->get_meta( '_dintero_checkout_line_id' );
+		}
+
+		if ( empty( $line_id ) ) {
+			// If the line_id is not set, we use the id as the line_id.
+			$line_id = $id;
+		}
```

Comment:
Question: Is this update backwards compatible to ensure that orders placed before the line id is added to shipping lines meta data can still be processed fully?